src/Mjolnir.ts

```diff
@@ -832,6 +834,35 @@ export class Mjolnir {
         await this.printBanlistChanges(changes, banList, true);
     }
 
+    private async handleConsequence(protection: IProtection, roomId: string, event: any, consequence: Consequence) {
```

Comment:
I like the idea of a `Consequence` not having to hold context of what caused the action because it's inherent; it pertains to whatever event triggered it. is the issue here just that we don't want to be passing `event: any` around?